pkg/apis/app/v2/kieapp_types.go

```diff
@@ -154,18 +154,39 @@ type KieServerSet struct {
 	// When set to true will configure the KIE Server image to disable all capabilities that are not related to decisions, Default to true for RHDM environments and false to RHPAM.
 	DecisionsOnly *bool `json:"decisionsOnly,omitempty"`
 	// Number of max KIE Executor sessions, it must be lower than the value of max-pool-size, by default is max-pool-size set to 60. Max pool size can be set by system property jboss.mdb.strict.max.pool.size (using javaOptsAppend: "-Djboss.mdb.strict.max.pool.size=40"), for more information see https://access.redhat.com/solutions/2955481.
-	KieExecutorMDBMaxSession *int `json:"kieExecutorMDBMaxSession,omitempty"`
+	KieExecutorMDBMaxSession *int              `json:"kieExecutorMDBMaxSession,omitempty"`
+	TerminationRoute         *TerminationRoute `json:"terminationRoute,omitempty"`
+}
+
+type TerminationRoute struct {
+	// Enable edge termination
+	EnableEdge bool `json:"enableEdge,omitempty"`
+	// Protocol to use, this could override the global disableSSL, http used with Edge enabled
+	// +kubebuilder:validation:Enum:=https;http
+	Protocol string `json:"protocol,omitempty"`
+	// +kubebuilder:validation:Enum:=passthrough;edge
+	// Passthrough is the default protocol, otherwise edge could be used.
+	Termination string `json:"termination,omitempty"`
```

Comment:
Is this needed together with EnableEdge property? I see that there can be some configuration conflicts. Or at least it may not be clear what all is needed and useful.

Reply:
- EnableEdge is the property from the cr to enable the edge property
- Protocol is used in the targetPort on the yaml (https/http)
- Termination is the termination on the yaml, (passthrough,edge)
- EnableTlsSection is the internal flag to enable/disable the tls section in the yaml, this can override the general disableSSL or copy the same value in case enableEdge is false
- Key, Certificate,CACErtificate are the sections onyaml for the private keys